content/docs/reference/glossary.md

```diff
@@ -6,54 +6,176 @@ redirectFrom:
+
+### Neon API
+
+The Neon RESTful Application Programming Interface. Any operation performed in the Neon Console can also be performed using the Neon API.
+
+### Neon Console
+
+A browser-based graphical interface for managing Neon projects and resources.
+
+### Neon user
+
+The user that registers and authenticates with Neon using a GitHub or Google account. Once authenticated, a Neon user can create and access projects, and manage users, databases, and other project resources.
+
+### Object storage
+
+Computer data storage that manages data as objects. Neon uses cloud object storage for long-term data storage.
+
+### Page
+
+An 8KB unit of data, which is the smallest unit that PostgreSQL uses for storing relations and indexes on disk. In Neon, a page is also the smallest unit of data that resides on a Pageserver. For information about PostgreSQL page format, see [Database Page Layout](https://www.postgresql.org/docs/14/storage-page-layout.html), in the _PostgreSQL Documentation_.
+
+### Pageserver
+
+A Neon architecture component that reads WAL records from Safekeepers to identify modified pages. The Pageserver accumulates and indexes incoming WAL records in memory and writes them to disk in batches. Each batch is written to an immutable file that is never modified after creation. Using these files, the Pageserver can quickly reconstruct any version of a page dating back to the user-defined retention period.
+
+The Pageserver uploads immutable files to cloud storage, which is the final, highly durable destination for data. Once a file is successfully uploaded to cloud storage, the corresponding WAL records can be removed from the Safekeepers.
+
+### Passwordless authentication
+
+The ability to authenticate without providing a password. Neon’s [Quick auth](#quick-auth) feature supports passwordless authentication.  
+
+### PostgreSQL user
+
+Two PostgreSQL users are created with each Neon project. The first is named for the registered Neon account and can be used to access the Neon project from a client. This user’s credentials can be managed and used for password-based `psql` authentication. The second user is the `web-access` system user, which is used by the SQL Editor and Neon’s (/docs/reference/glossary/#quick-auth) feature. The `web-access` user is system managed. It cannot be modified, removed, or used in other authentication scenarios.
+
+Additional PostgreSQL users can be created in the Neon Console.
+
+### Project
+
+A collection of PostgreSQL databases, PostgreSQL users, and other project resources and settings. A project contains a Compute with a PostgreSQL server as well as storage for the project data.
+
+### Proxy
+
+A multitenant service that accepts and handles connections from clients that use the PostgreSQL protocol.
```

Comment:
i suggest to refer to Neon components as components rather than services to avoid confusion. e.g.

```suggestion
A component of Neon, a multitenant service that accepts and handles connections from clients that use the PostgreSQL protocol.
```

can we also add the links to github?

Reply:
Revised to use "component"